[8.0][MIG] oauth provider backport from 9.0

hhgabelgaard opened the pull request:
oauth_provider backport from 9.0:

- `<odoo> -> <openerp><data>`
- crsf handling removed
- website added to http routes
- Exceptions: UserInfo -> Warning

Scope preserved from when swicthing authentification code to token
Extension point for additional userinfo response

Commit:
[ADD] Add oauth_provider module

Commit:
[ADD] Add oauth_provider module

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

hughesbm committed:
[ADD] module_auto_update: Create module (#882)

* [IMP] module_auto_update: Create new module
* Add checksum_dir and checksum_installed fields to ir.module.module
  * Add checksum_dir to compute current checksum of module directory
    in addons path
  * Add checksum_installed to store checksum of module directory
    when module was last installed or upgraded
* Use checksumdir Python library to compute module directory sha1
  hashes, ignoring pyc and pyo extensions
* Extend update_list method to compare modules' checksum_dir and
  checksum_installed, then change state of modules with differing
  checksums to 'to upgrade'
* Replace Apps/Updates menu item with menu item of same name, which
  updates apps list and displays tree view of ir.module.module
  records with state 'to upgrade'
* Extend create and write methods to store computed checksum_dir as
  checksum_installed during module installation and upgrade, and
  set checksum_installed to False on uninstall
* Use context to stop checksum_installed from being updated during
  upgrade/uninstall cancellation
* Add cron job to periodically check for module upgrades by
  comparing checksums, then perform any available upgrades
* Extend upgrade_module method (called by cron and 'Apply Scheduled
  Upgrades' menu item) to call update_list
* Add post_init_hook to store checksum_installed of existing
  modules
* Add test coverage

* [FIX] module_auto_update: Fix test broken by changes
* Use dummy module to test update_list method instead of
  module_auto_update

oca-transbot committed:
OCA Transbot updated translations from Transifex

yajo committed:
[MIG][module_auto_update] Backport from v10

yajo committed:
[REF][module_auto_update] Recompute dir hashes only when needed

By removing the recomputation from `update_list` we get faster CLI module upgrades and it only performs the autoupdate when using the autoupdate wizard or cron.

yajo committed:
[FIX][module_auto_update] Pass tests if addon is in readonly directory

yajo committed:
[FIX][module_auto_update] Set dependencies to upgrade

pedrobaeza committed:
[FIX] module_auto_update: Missing author

oca-transbot committed:
OCA Transbot updated translations from Transifex

yajo committed:
[FIX][module_auto_update] Record base addon checksum (#948)

yajo committed:
[FIX][module_auto_update] Always store changes in lower graphs (#953)

Commit:
[FIX] oauth_provider: Fix tests when using last oauthlib version

Commit:
[FIX] oauth_provider: Fix tests when using last oauthlib version

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
[REF] module_auto_update: Step 2, add new API

This code comes from the module_checksum_upgrade proposal
at https://github.com/OCA/server-tools/pull/1176.

* [ADD] module_checksum_upgrade

It provides the core mechanism of module_auto_update without
the cron nor any change to the standard upgrade mechanism.
Instead it provides an API on which module_auto_update can build,
as well as a method which can be called from a script to run
the upgrade of modules for which the checksum has changed.

* [IMP] refactor module_auto_update

Make it depend on module_checksum_upgrade which provides
the core mechanisms of managing the checksums. module_auto_update
makes it automatic.

* [IMP] module_checksum_upgrade: better exclusion mechanism

Ignore files based on exclude patterns.
Ignore uninstalled languages.
Better default for patterns to ignore (*.pyc,*.pyo,*.pot,static/*)

For better control on the hashing mechanism implement our own:
it's quite easy, and the checksumdir module used previously had
no test.

* [MIG] module_auto_update: adapt to new checksum mechanism

* [IMP] module_checksum_upgrade: raise in case of
 incomplete upgrade

* [IMP] module_checksum_upgrade: improve default exclusion
 pattern

* [IMP] module_checksum_upgrade: control translations
 overwrite

* [IMP] module_checksum_upgrade: one more test

* [IMP] module_checksum_upgrade: credits [ci skip]

yajo committed:
[REF] module_auto_update: Step 1, move all deprecated stuff to deprecated files

- Files are clearly suffixed with `_deprecated` so we know those features have no support nor migrations.
- Views are removed, since updating from UI was too buggy to support it anymore.

yajo committed:
[REF] module_auto_update: Step 3, backwards compatibility

The previous implementation of this addon proved being extremely buggy:

- It supplied out of the box a enabled cron to update Odoo that didn't restart the server, which possibly meant that upgrades broke things.
- It overloaded standard Odoo upgrade methods that made i.e. installing an addon sometimes forced to upgrade all other addons in the database.
- The checksum system wasn't smart enough, and some files that didn't need a module upgrade triggered the upgrade.
- It was based on a dirhash library that was untested.
- Some updates were not detected properly.
- Storing a column into `ir.module.module` sometimes forbids uninstalling the addon.

Thanks to Stéphane Bidoul (ACSONE), now we have new methods to perform the same work in a safer and more stable way.

All I'm doing here is:

- Cron is disabled by default.
- Installed checksums are no longer saved at first install.
- Old installations should keep most functionality intact thanks to the migration script.
- Drop some duplicated tests.
- Allow module uninstallation by pre-removing the fields from ir.mode.model.
- When uninstalling the addon, the deprecated features will get removed for next installs always.

Besides that, fixes for the new implementation too:

- When uninstalling the addon, we remove the stored checksum data, so further installations work as if the addon was installed from scratch.

sbidoul committed:
[MIG] module_auto_update from 9 to 8

hhgabelgaard committed:
oauth_provider backport from 9.0

<odoo> -> <openerp><data>
crsf handling removed
website added to http routes
Avoid conflicts on relations name
Exceptions
Runbot link

Extension point for additional userinfo response

hhgabelgaard committed:
oauth_provider backport from 9.0

<odoo> -> <openerp><data>
crsf handling removed
website added to http routes
Avoid conflicts on relations name
Exceptions
Runbot link

Extension point for additional userinfo response

pedrobaeza commented:
Please squash together adjacent "OCA Transbot..." commits using this guide: https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests

JayVora-SerpentCS commented on oauth_provider/__openerp__.py:
```diff
@@ -0,0 +1,28 @@
+{
+    'name': 'OAuth Provider',
+    'summary': 'Allows to use Odoo as an OAuth2 provider',
+    'version': '9.0.1.0.0',
```
8.0.1.0.0 please

hhgabelgaard committed:
requirements: oauthlib added

hhgabelgaard committed:
requirements: oauthlib added

hhgabelgaard committed:
pylint

hhgabelgaard committed:
pylint

hhgabelgaard reviewed (commented):
` cr.execute('ALTER TABLE res_users ADD COLUMN oauth_identifier varchar')`
in oauth_provider/__init.py__
Why is Travis failing over this line? Is it installing the module twice?
Neither OCA's Runbot or my own fails over this

pedrobaeza commented:
Because the column already exists. Is it created by other module? Maybe you should check if the column exists.

There's a flake8 error too:
```
oauth_provider/models/res_users.py:26:18: W291 trailing whitespace
```

Garamotte commented:
The obvious issue is that the init hook must check if the column exists before trying to create it.
But there is also [another issue](https://github.com/OCA/server-tools/issues/1202), IMO due to the database_cleanup module (to be fixed in another PR).

hhgabelgaard committed:
Check if oauth_identifier already exist. 
In that case, skip the initializing

hhgabelgaard committed:
Check if oauth_identifier already exist. 
In that case, skip the initializing

hhgabelgaard committed:
_fields is not initiliazed at this point

hhgabelgaard committed:
_fields is not initiliazed at this point

hhgabelgaard committed:
Test

hhgabelgaard committed:
Test

hhgabelgaard committed:
test

hhgabelgaard committed:
test

hhgabelgaard commented:
Hi @sylvain-garancher 
Thanks for the analysis in #1202 - That's explain it, and yes, the obvious solution is to check for existence in the pre_init hook. That's now implemented, and the log lines (to be removed)  shows that pre_init is called twice, but the column adding is only done once. So far so good.

But now I'm getting errors from auth_brute_force: [KeyError: 'res.authentication.attempt'](https://travis-ci.org/OCA/server-tools/jobs/358531175#L1710) in line 1710 in the Travis log - But auth_brute_force is installed earlier - in line 1228: [2018-03-26 18:10:00,539 5101 INFO openerp_test openerp.modules.module: module auth_brute_force: creating or updating database tables](https://travis-ci.org/OCA/server-tools/jobs/358531175#L1228)

Could that be due to the same issue? Are the tests being run on "half a registry"?
I'm not familiar with the auth_brute_force module, and it doesn't exist in 9.0 so I can't exclude some kind of "incompabilty" between them.

And why is it only on Travis? Isn't the same tests been run under the Runbot? 
Should I add EXCLUDE="database_cleanup" to .travis.yml ?

Transbot commits is squashed, flake/lint tests is green

yajo committed:
[FIX] module_auto_update: Add .pyo sample file (#1205)

Without this patch, if your tests are run under a `PYTHONOPTIMIZE=2` precompiled environment, they'd fail with this error because a new `.pyo` file would be created.:

    FAIL: test_basic (openerp.addons.module_auto_update.tests.test_addon_hash.TestAddonHash)
    Traceback (most recent call last):
    `   File "/opt/odoo/auto/addons/module_auto_update/tests/test_addon_hash.py", line 41, in test_basic
    `     'static/src/some.js',
    ` AssertionError: Lists differ: ['README.rst', 'data/f1.xml', ... != ['README.rst', 'data/f1.xml', ...
    `
    ` First differing element 13:
    ` models/stuff.pyo
    ` static/src/some.js
    `
    ` First list contains 1 additional elements.
    ` First extra element 14:
    ` static/src/some.js
    `
    `   ['README.rst',
    `    'data/f1.xml',
    `    'data/f2.xml',
    `    'i18n/en.po',
    `    'i18n/en_US.po',
    `    'i18n/fr.po',
    `    'i18n/fr_BE.po',
    `    'i18n/test.pot',
    `    'i18n_extra/en.po',
    `    'i18n_extra/fr.po',
    `    'i18n_extra/nl_NL.po',
    `    'models/stuff.py',
    `    'models/stuff.pyc',
    ` -  'models/stuff.pyo',
    `    'static/src/some.js']
    Ran 3 tests in 0.005s
    FAILED

With this patch, the `.pyo` file is included, so tests will pass anywhere.

pedrobaeza committed:
Merge pull request #1198 from acsone/8.0-module_auto_update-sbi

[MIG][8.0] module_auto_update backport from 9

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] setup.py

hbrunn committed:
[ADD] create an xmlid for mapped records too (#1059)

* [ADD] create an xmlid for mapped records too
* [IMP] filter out unknown fields
* [FIX] warning about exception handling

hhgabelgaard committed:
Exclude database_cleanup from tests - fixes #1202

hhgabelgaard committed:
Exclude database_cleanup from tests - fixes #1202

OCA-git-bot committed:
[ADD] setup.py [ci skip]

sbidoul committed:
module_auto_update: check a post condition

sbidoul committed:
module_auto_update: bump version

OCA-git-bot committed:
[UPD] addons table in README.md [ci skip]

hhgabelgaard committed:
Merge branch '8.0-oca-oauth_provider-backport' of https://github.com/steingabelgaard/server-tools.git into 8.0-oca-oauth_provider-backport
